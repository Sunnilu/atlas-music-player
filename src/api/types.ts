// src/api/types.ts
export interface Song {
  url: any;
  audio: any;
  image: string;
  author: string;
  id: number;
  title: string;
  artist: string;
  genre: string;
  duration: string;
  cover: string;
  song: string;
}

export interface ApiResponse<T> {
  status: 'success' | 'error';
  data?: T;
  error?: string;
}