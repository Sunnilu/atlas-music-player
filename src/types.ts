// src/types.ts
export interface Song {
  id: number;
  title: string;
  author: string;      // 🔁 renamed from `artist`
  duration: string;
  image: string;
  lyrics?: string;
  audio: string;
}
export {};
