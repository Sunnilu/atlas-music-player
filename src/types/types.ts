// src/types.ts
export interface Song {
    id: number;
    title: string;
    artist: string;
    duration: string;
    image: string;   // URL for album artwork
    lyrics?: string; // Optional lyrics field
  }
  