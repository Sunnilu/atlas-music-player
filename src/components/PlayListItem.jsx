// src/components/PlayListItem.jsx
import React from 'react';
import PropTypes from 'prop-types';
import clsx from 'clsx';

export default function PlayListItem({ song, isSelected, onClick }) {
  return (
    <div
      className={clsx(
        'flex justify-between items-center p-4 rounded-lg cursor-pointer transition-colors',
        isSelected ? 'bg-gray-200' : 'hover:bg-gray-100'
      )}
      onClick={onClick}
    >
      <div>
        <p className="text-sm font-semibold text-gray-900">{song.title}</p>
        <p className="text-xs text-gray-500">{song.artist}</p>
      </div>
      <span className="text-sm text-gray-700">{song.duration}</span>
    </div>
  );
}

PlayListItem.propTypes = {
  song: PropTypes.shape({
    title: PropTypes.string.isRequired,
    artist: PropTypes.string.isRequired,
    duration: PropTypes.string.isRequired,
  }).isRequired,
  isSelected: PropTypes.bool,
  onClick: PropTypes.func,
};
