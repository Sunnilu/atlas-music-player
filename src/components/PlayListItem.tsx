// src/components/PlayListItem.tsx
import React from 'react';
import { Song } from '../types';

interface PlayListItemProps {
  song: Song;
  isSelected: boolean;
  onClick: () => void;
}

const PlayListItem: React.FC<PlayListItemProps> = ({ song, isSelected, onClick }) => {
  return (
    <div
      onClick={onClick}
      className={`cursor-pointer px-4 py-3 rounded-lg transition-all ${
        isSelected
          ? 'bg-primary text-white font-semibold shadow-md'
          : 'hover:bg-accent hover:text-white'
      }`}
    >
      <div className="flex justify-between items-center">
        <div>
          <p className="text-sm">{song.title}</p>
          <p className="text-xs opacity-80">{song.artist}</p>
        </div>
        <span className="text-xs">{song.duration}</span>
      </div>
    </div>
  );
};

export default PlayListItem;
