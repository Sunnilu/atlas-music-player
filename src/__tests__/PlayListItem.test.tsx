// src/__tests__/PlayListItem.test.tsx
import { describe, it, expect } from 'vitest';
import renderer from 'react-test-renderer';
import PlayListItem from '../components/PlayListItem';
import { Song } from '../types';

const mockSongBase: Song = {
    id: 1,
    title: 'Moonlight Sonata',
    artist: 'Beethoven',
    author: 'L. van Beethoven',
    duration: '4:35',
    image: '',
    audio: '',
    lyrics: '',
    cover: undefined
};

describe('PlayListItem Snapshot', () => {
  it('renders as selected', () => {
    const tree = renderer
      .create(
        <PlayListItem song={mockSongBase} isSelected={true} onClick={() => {}} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders as not selected', () => {
    const tree = renderer
      .create(
        <PlayListItem song={mockSongBase} isSelected={false} onClick={() => {}} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with short duration', () => {
    const shortSong: Song = { ...mockSongBase, duration: '45' }; // 0:45 formatted
    const tree = renderer
      .create(
        <PlayListItem song={shortSong} isSelected={false} onClick={() => {}} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
