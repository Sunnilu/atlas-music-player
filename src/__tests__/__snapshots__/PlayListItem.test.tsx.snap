// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayListItem Snapshot > renders as not selected 1`] = `null`;

exports[`PlayListItem Snapshot > renders as selected 1`] = `null`;

exports[`PlayListItem Snapshot > renders with short duration 1`] = `null`;
